src/sentry/api/endpoints/organization_metrics.py

```diff
@@ -154,7 +157,31 @@ class OrganizationMetricsDataEndpoint(OrganizationEndpoint):
     owner = ApiOwner.TELEMETRY_EXPERIENCE
     default_per_page = 50
 
-    def get(self, request: Request, organization) -> Response:
+    def _new_get(self, request: Request, organization) -> Response:
+        projects = self.get_projects(request, organization)
+
+        # We first parse the interval and date, since this is dependent on the query params.
+        interval = parse_stats_period(request.GET.get("interval", "1h"))
+        interval = int(3600 if interval is None else interval.total_seconds())
+        start, end = get_date_range_from_params(request.GET)
+
+        # We then run the query and inject directly the field, query and groupBy, since they will be parsed
+        # internally.
+        results = run_metrics_query(
+            fields=request.GET.getlist("field", []),
+            query=request.GET.get("query"),
+            group_bys=request.GET.getlist("groupBy"),
+            interval=interval,
+            start=start,
+            end=end,
+            use_case_id=get_use_case_id(request),
```

Comment:
Do not pass the `use_case_id` anymore, instead infer it _internally_ from the parsed `Timeseries` since the MRIs by design specify the namespace (aka use case). This should eventually be handled by the query layer.